gold/testsuite/Makefile.am

```diff
@@ -4334,6 +4338,9 @@ nanomips_move_balc_1_out_of_range_pcrel: nanomips_move_balc_out_of_range_pcrel.o
 nanomips_move_balc_2_out_of_range_pcrel: nanomips_move_balc_out_of_range_pcrel.o ../ld-new
 	../ld-new --section-start .text=0x1000 --section-start .foo=0x2001020 -o $@ $<
 
+nanomips_long_balc_bc_offset: nanomips_long_balc_bc_offset.o ../ld-new
+	../ld-new -expand-long-balc-bc --section-start .text=0x1000 --section-start .foo=0xff1008 -o $@ $<
```

Comment:
If would be instructive to add a test-case for the largest value that does not trigger the expansion. What you would find is 

1. that the range used here (0xff0000), is short of the actual range (0x1ffffff) by a whole bit position
2. The only reason this test passes is because the linker is performing the transformation regardless of the actual address delta whenever you pass -expand-ling-balc-bc

Likewise, tests for the -ve range would be good.

Reply:
Oh, the test actually does not pass, a copy-paste mistake (wrong range used in it). I will fix this along with addressing the comments.

Reply:
Actually, the test passes as written, but it shouldn't. I think this indicates that the checking expression in the code (the one with signed/unsigned comparison) is not working as you'd expect.

Reply:
Added. Thanks.